src/js/App.js

```diff
@@ -50,18 +51,61 @@ window.App = Backbone.View.extend({
     },
 
     fadeInListView: function() {
+        this.regionListView = new RegionListView({ model: this.model });
+        this.regionListView.render();
         this.regionListView.$el.fadeIn(200);
     },
 
+    animateDetailsViewReplacingList: function() {
+        var self = this;
+        var selectedRegion = this.model.get('selectedRegion').get('title');
+        var listItems = $('#region-location-list').children();
+        var $sidebarInnerEl = $('.sidebar--inner');
+        var promise = $.Deferred();
+        var $selectedListItem;
+
+        promise.then(function() {
+            self.regionListView.$el.hide();
+            self.regionListView.$el.empty();
+            self.regionDetailsView.render();
+            self.regionDetailsView.$el.show();
+        });
+
+        $.each(listItems, function(index, listItem) {
+            if (listItem.id !== selectedRegion) {
+                $(listItem).delay(200 * index).fadeTo(300, 0);
+            } else {
+                $selectedListItem = $(listItem);
+            }
+        });
+
+        // If $selectedListItem's unset, toggle views without animating
+        if (!$selectedListItem) {
+            promise.resolve();
+            return _.noop();
+        }
+
+        $selectedListItem
+            .delay(listItems.length * 200)
+            .queue(function(next) {
+                var top = $sidebarInnerEl.offset().top - $selectedListItem.offset().top;
+                $selectedListItem.animate({
+                    top: top + 'px',
+                    margin: '0 -3rem 0 -3rem'
+                }, 500, promise.resolve).fadeOut(500);
+                next();
+            });
+
+        return _.noop();
```

Comment:
Consider adding a comment here explaining why we return this.

Reply:
Sounds good. I'll make a note that this was done to ensure the function has the same return type whether or not the animation code gets executed.